File: SDKs/bigredcloud.com/v1/python/src/sdk/models/shared/customerdto.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class CustomerDto:
    account_name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accountName') }})
    account_number: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('accountNumber') }})
    address: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('address') }})
    auth_code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('authCode') }})
    bank: Optional[EftBankDto] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('bank') }})
    business_identifier_code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('businessIdentifierCode') }})
    code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('code') }})
    contact: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('contact') }})
    delivery: Optional[List[str]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('delivery') }})
    e_ft_reference: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('eFTReference') }})
    email: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('email') }})
    fax: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fax') }})
    id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('id') }})
    international_bank_account_number: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('internationalBankAccountNumber') }})
    ledger_balance: Optional[float] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ledgerBalance') }})
    mobile: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('mobile') }})
    name: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    opening_balance: Optional[OwnerOpeningBalanceInPeriodsDto] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('openingBalance') }})
    opening_balances: Optional[List[OwnerOpeningBalanceDto]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('openingBalances') }})
    our_code: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ourCode') }})
    owner_type_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ownerTypeId') }})
    phone: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('phone') }})
    timestamp: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('timestamp') }})
    vat_analysis_type_id: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('vatAnalysisTypeId') }})
    vat_reg: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('vatReg') }})
    vat_type: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('vatType') }})
    

File: SDKs/bigredcloud.com/v1/typescript/src/sdk/models/shared/customerdto.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
import { EftBankDto } from "./eftbankdto";
import { OwnerOpeningBalanceInPeriodsDto } from "./owneropeningbalanceinperiodsdto";
import { OwnerOpeningBalanceDto } from "./owneropeningbalancedto";


export class CustomerDto extends SpeakeasyBase {
  @Metadata({ data: "json, name=accountName" })
  accountName?: string;

  @Metadata({ data: "json, name=accountNumber" })
  accountNumber?: string;

  @Metadata({ data: "json, name=address" })
  address?: string[];

  @Metadata({ data: "json, name=authCode" })
  authCode?: string;

  @Metadata({ data: "json, name=bank" })
  bank?: EftBankDto;

  @Metadata({ data: "json, name=businessIdentifierCode" })
  businessIdentifierCode?: string;

  @Metadata({ data: "json, name=code" })
  code?: string;

  @Metadata({ data: "json, name=contact" })
  contact?: string;

  @Metadata({ data: "json, name=delivery" })
  delivery?: string[];

  @Metadata({ data: "json, name=eFTReference" })
  eFtReference?: string;

  @Metadata({ data: "json, name=email" })
  email?: string;

  @Metadata({ data: "json, name=fax" })
  fax?: string;

  @Metadata({ data: "json, name=id" })
  id?: number;

  @Metadata({ data: "json, name=internationalBankAccountNumber" })
  internationalBankAccountNumber?: string;

  @Metadata({ data: "json, name=ledgerBalance" })
  ledgerBalance?: number;

  @Metadata({ data: "json, name=mobile" })
  mobile?: string;

  @Metadata({ data: "json, name=name" })
  name?: string;

  @Metadata({ data: "json, name=openingBalance" })
  openingBalance?: OwnerOpeningBalanceInPeriodsDto;

  @Metadata({ data: "json, name=openingBalances", elemType: shared.OwnerOpeningBalanceDto })
  openingBalances?: OwnerOpeningBalanceDto[];

  @Metadata({ data: "json, name=ourCode" })
  ourCode?: string;

  @Metadata({ data: "json, name=ownerTypeId" })
  ownerTypeId?: number;

  @Metadata({ data: "json, name=phone" })
  phone?: string;

  @Metadata({ data: "json, name=timestamp" })
  timestamp?: string;

  @Metadata({ data: "json, name=vatAnalysisTypeId" })
  vatAnalysisTypeId?: number;

  @Metadata({ data: "json, name=vatReg" })
  vatReg?: string;

  @Metadata({ data: "json, name=vatType" })
  vatType?: number;
}
